uikit/common/src/main/java/com/sygic/ui/common/behaviors/BottomSheetBehaviorWrapper.kt

```diff
@@ -0,0 +1,82 @@
+package com.sygic.ui.common.behaviors
+
+import android.view.View
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import java.lang.ref.WeakReference
+
+class BottomSheetBehaviorWrapper(private val behavior: BottomSheetBehavior<View>) {
+
+    interface StateListener {
+        fun onStateChanged(@BottomSheetBehavior.State newState: Int)
+    }
+
+    interface SlideListener {
+        fun onSlide(slideOffset: Float)
+    }
+
+    @BottomSheetBehavior.State
+    var state: Int
+        get() = behavior.state
+        set(value) {
+            behavior.state = value
+        }
+
+    var isHideable: Boolean
+        get() = behavior.isHideable
+        set(value) {
+            behavior.isHideable = value
+        }
+
+    var peekHeight: Int
+        get() = behavior.peekHeight
+        set(value) {
+            behavior.peekHeight = value
+        }
+
+    private val stateListeners = LinkedHashSet<WeakReference<StateListener>>()
+    private val slideListeners = LinkedHashSet<WeakReference<SlideListener>>()
```

Comment:
nebude problem, ze inline callbacky nam budu miznut? ci?

Reply:
FYI: co sa tyka tej synchornizacie, existuje este skupina metod `Collections.synchronized...`, ktore sa daju pouzit. Mozes len tak mrknut ockom ;)

Reply:
jj, ale už víš proč jsem to tak udělal :-D navíc je to RestrictTo LIBRARY_GROUP ;)